crates/primitives/starknet/src/execution/contract_class_wrapper.rs

```diff
@@ -44,104 +47,191 @@ impl<'de> Deserialize<'de> for ProgramWrapper {
-                        )
-                    },
-                )),
-            }
-            .into(),
-        )) // TODO (Greg) handle v1
+        match value {
+            ContractClassWrapper::V0(class) => Self::V0(ContractClassV0(
+                ContractClassV0Inner {
+                    program: class.program.into(),
+                    // Convert EntrypointMapWrapper to HashMap<EntryPointType, Vec<EntryPoint>>
+                    entry_points_by_type: HashMap::from_iter(class.entry_points_by_type.0.iter().clone().map(
+                        |(entrypoint_type, entrypoints)| {
+                            (
+                                entrypoint_type.clone().into(),
+                                entrypoints.clone().into_iter().map(|val| val.into()).collect(),
+                            )
+                        },
+                    )),
+                }
+                .into(),
+            )),
+            ContractClassWrapper::V1(class) => Self::V1(ContractClassV1(
+                ContractClassV1Inner {
+                    program: class.program.into(),
+                    // Convert EntrypointMapWrapper to HashMap<EntryPointType, Vec<EntryPoint>>
+                    entry_points_by_type: HashMap::from_iter(class.entry_points_by_type.0.iter().clone().map(
+                        |(entrypoint_type, entrypoints)| {
+                            (
+                                entrypoint_type.clone().into(),
+                                entrypoints.clone().into_iter().map(|val| val.into()).collect(),
+                            )
+                        },
+                    )),
+                    hints: class.hints.0.clone(),
+                }
+                .into(),
+            )),
+        }
     }
 }
 
 impl From<ContractClass> for ContractClassWrapper {
     fn from(value: ContractClass) -> Self {
         match value {
-            ContractClass::V0(class) => Self {
+            ContractClass::V0(class) => Self::V0(ContractClassV0Wrapper {
                 program: class.program.clone().into(),
-                entry_points_by_type: EntrypointMapWrapper(unsafe {
+                entry_points_by_type: EntrypointMapV0Wrapper(unsafe {
                     mem::transmute::<
                         HashMap<EntryPointType, Vec<EntryPoint>>,
-                        HashMap<EntryPointTypeWrapper, Vec<EntryPointWrapper>>,
+                        HashMap<EntryPointTypeWrapper, Vec<EntryPointV0Wrapper>>,
                     >(class.entry_points_by_type.clone())
                 }),
-            },
-            _ => Self::default(), // TODO (Greg) handle v1
+            }),
+            ContractClass::V1(class) => Self::V1(ContractClassV1Wrapper {
+                program: class.program.clone().into(),
+                entry_points_by_type: EntrypointMapV1Wrapper(unsafe {
+                    mem::transmute::<
```

Comment:
hm not a huge fan of this, we want to copy a hashmap to another right ?
Did we benchmark the cost of a *safe* implementation ?

Reply:
I didn't benchmark it but this wrapper will be removed entirely and the `Encode`, `Decode` traits will be implemented directly within `ContractClass`